src/components/OntologyBrowser/store.ts

```diff
@@ -19,10 +19,69 @@ export type TreeNode = {
   exactTagCount?: number;
   hidden?: boolean;
   depth?: number;
+  disabled?: boolean;
 };
 
 const dotToUnderscore = (str: string) => str.replace('.', '__');
 
+const termRegex = new RegExp('[^-]+$');
+
+//pattern of term with escaped parentheses
+const termPattern = (term: string) => new RegExp(`.*${term?.replace(/(?=[()])/g, '\\')}$`);
+
+export const removeSameTerms = (selectedKeys: string[], targetKeys: string[]) => {
+  let updatedTargetKeys = targetKeys;
+
+  selectedKeys.forEach((t) => {
+    const match = t.match(termRegex);
+    if (match) {
+      const term = match.pop();
+
+      if (term) {
+        const pattern = termPattern(term);
+        updatedTargetKeys = updatedTargetKeys.filter((t) => !pattern.test(t));
+      }
+    }
+  });
+  return [...updatedTargetKeys, ...selectedKeys];
+};
+
+export const selectSameTerms = (selectedKeys: string[], tree: TreeNode[] | undefined) => {
+  let enhancedSelectedKeys: string[] = [];
+  if (tree) {
```

Comment:
why not exit quickly => if(!tree) { return [] } it would remove 1 level of indentation